cli.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+"""
+This example uses docopt with the built in cmd module to demonstrate an
+interactive command application.
+Usage:
+    smart-library desk --action=<action> --name=<name> --i2c_address=<i2c_address>
+    smart-library (-i | --interactive)
+    smart-library (-h | --help | --version)
+Options:
+    -i, --interactive  Interactive Mode
+    -h, --help  Show this screen and exit.
+"""
+
+import sys, cmd, re
+from docopt import docopt, DocoptExit
+from services.desk_handler_service import DeskHandlerService
+from colorama import Fore, Style
+
+def docopt_cmd(func):
+    """
+    This decorator is used to simplify the try/except block and pass the result
+    of the docopt parsing to the called action.
+    """
+    def fn(self, arg):
+        try:
+            opt = docopt(fn.__doc__, self.format_args(arg))
+        except DocoptExit as e:
+            # The DocoptExit is thrown when the args do not match.
+            # We print a message to the user and the usage block.
+
+            print(Fore.RED + 'Whoops! Something went wrong. Please try again.')
+            print(e + Style.RESET_ALL)
+            return
+
+        except SystemExit:
+            # The SystemExit exception prints the usage for --help
+            # We do not need to do the print here.
+
+            return
+
+        return func(self, opt)
+
+    fn.__name__ = func.__name__
+    fn.__doc__ = func.__doc__
+    fn.__dict__.update(func.__dict__)
+    return fn
+
+
+class MyInteractive (cmd.Cmd):
+    intro = 'Welcome to the Smart Library command line interface! (type help for a list of commands.)'
+    prompt = '(smart-library) '
+    file = None
+
+    @docopt_cmd
+    def do_desk(self, arg):
+        """
+        Usage: desk --action=<action> --name=<name> --i2c_address=<i2c_address>
+        
+        Options:
+        --action=<action> Action on a desk [create]
+        --name=<name> Name of desk to execute action on
+        --i2c_address=<i2c_address> Address on the I2C bus
+        """
+
+        action = arg['--action']
+        name = arg['--name']
+        i2c_address = arg['--i2c_address']
+
+        if action == 'create': 
```

Comment:
`docopt` seems to not have support for commands in the format of `desk create ...` so I had to add an `action` flag to mimic the behaviour I was looking for

Reply:
FYI, this sort of behaviour is available in the [cmd library](https://www.google.ca/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0ahUKEwjw3qqTqdvYAhWQ2YMKHXJQDNoQFggpMAA&url=https%3A%2F%2Fdocs.python.org%2F2%2Flibrary%2Fcmd.html&usg=AOvVaw0dcfi7gbze8t0PLG8gRhRg)

Reply:
FYI [cmd](https://docs.python.org/3/library/cmd.html) does this

Reply:
Yup! We're using it in conjunction with `docopt`. I still couldn't found a way to cleanly point `desk create ...` to a `do_*` method hence why I went with `--action` instead

Reply:
I think this [github repo](https://github.com/rgreinho/docopt-subcommands-example) might be addressing this issue. Might be something to consider in a future PR. Wouldn't consider it on the critical path though, this looks good already.

Reply:
Nice! I'll take a look after we start doing the improvements.